File: app/src/main/java/com/example/ffmpegcmd/ffmpegjava/FFmpegHandler.java
package com.example.ffmpegcmd.ffmpegjava;

import android.os.Handler;

/**
 * Created by Vashon on 2021/8/2.
 */
public class FFmpegHandler {

    public static final int STATE_START = 615;
    public static final int STATE_MESSAGE = 616;
    public static final int STATE_PROGRESS = 617;
    public static final int STATE_FINISH = 618;

    private Handler mHandler;

    public FFmpegHandler(Handler mHandler) {
        this.mHandler = mHandler;
    }

    public void executeFFmpegCmd(String[] commands) {
        if (mHandler != null) {
            FFmpegCmd.getInstance().execute(commands, new OnHandleListener() {
                @Override
                public void onStart() {
                    mHandler.obtainMessage(STATE_START).sendToTarget();
                }

                @Override
                public void onMessage(String message) {
                    mHandler.obtainMessage(STATE_MESSAGE, message).sendToTarget();
                }

                @Override
                public void onProgress(int position, int duration) {
                    mHandler.obtainMessage(STATE_PROGRESS, position, duration).sendToTarget();
                }

                @Override
                public void onFinish() {
                    mHandler.obtainMessage(STATE_FINISH).sendToTarget();
                }
            });
        }
    }
}

File: app/src/main/java/com/example/ffmpegcmd/ffmpeg/FFmpegHandler.kt
package com.example.ffmpegcmd.ffmpeg

import android.os.Handler


class FFmpegHandler(handler: Handler) {

    companion object{
        const val MSG_START: Int = 10
        const val MSG_INFO: Int = 10
        const val MSG_PROGRESS: Int = 10
        const val MSG_FINISH: Int = 10
    }

    private var mHandler: Handler? = null

    init {
        mHandler = handler
    }


    /**
     * 执行ffmpeg命令
     * @param 命令字符串
     * */
    fun executeFFmpegCmd(array: Array<String?>?){
        FFmpegCmd.instance?.execute(array, object :OnHandleListener{
            override fun onStart() {
                mHandler?.obtainMessage(MSG_START)?.sendToTarget()
            }

            override fun onMessage() {
                mHandler?.obtainMessage(MSG_INFO)?.sendToTarget()
            }

            override fun onProgress() {
                mHandler?.obtainMessage(MSG_PROGRESS)?.sendToTarget()
            }

            override fun onFinish() {
                mHandler?.obtainMessage(MSG_FINISH)?.sendToTarget()
            }

        })
    }
}